### ISSUE TITLE
create-remix@latest results in hydration errors on unmodified build

### ISSUE THREAD

**Issue by wulfmann** [remix-run/remix#5416]
### What version of Remix are you using?

1.12.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

```
% npx create-remix@latest repro
? What type of app do you want to create? Just the basics
? Where do you want to deploy? Choose Remix App Server if you're unsure; it's easy to change deployment targets. Vercel
? TypeScript or JavaScript? TypeScript
? Do you want me to run `npm install`? Yes
```

Then:

```
npm run dev
```

![console screenshot](https://github.com/wulfmann/remix-repro/blob/master/remix-repro-console-screenshot.png?raw=true)

Repro: https://github.com/wulfmann/remix-repro

### Expected Behavior

A fresh bare build does not log hydration errors.

### Actual Behavior

Hydration errors are logged complaining about a missing closing `<head>` tag.

**Comment by manurana** [comment]
OK, I think I figured this one out
I had to disable the Apollo Client Devtools extension to fix this error
This was observed on Firefox Developer Edition 110, as well as Chrome Canary 112

**Comment by wulfmann** [comment]
Sorry I should have included I was on chrome. I also had that extension and disabling it indeed seems to have fixed this issue. Nice find!

Is this an issue with the extension then?

**Comment by frontendphil** [comment]
Mind. Blown. 🤯 Sorry for the spam, but thanks SO MUCH! That extension caused me hours of looking through the code why stuff would be different.

**Comment by clgeoio** [comment]
This issue has been seen before - there's a proposal about it here: https://github.com/remix-run/remix/discussions/5244

**Comment by l-portet** [comment]
Had this issue happening with the Meteor devtools extension too (Chrome 111).

**Comment by nichtsam** [comment]
Same issue, disabling all extensions fixed it, I'm guessing the scripts injection of extensions caused this?

```zsh
pnpm create remix@latest
# node v18.15.0
# Brave 1.50.114 Chromium: 112.0.5615.49 (arm64)
```

**Comment by brophdawg11** [comment]
I believe this is a duplicate of the #5244 and will be (hopefully) addressed when React 18.3 is released.  See https://github.com/facebook/react/issues/24430#issuecomment-1440427646 for more information.

I'm going to close this out as there's not much Remix can do it seems.
